engines/bladerunner/bladerunner.cpp

```diff
@@ -324,8 +324,22 @@ Common::Error BladeRunnerEngine::run() {
 		return Common::Error(Common::kNoGameDataFoundError, missingFileStr);
 	}
 
-	Graphics::PixelFormat format = screenPixelFormat();
-	initGraphics(640, 480, &format);
+	Common::List<Graphics::PixelFormat> tryModes = _system->getSupportedFormats();
+	for (Common::List<Graphics::PixelFormat>::iterator g = tryModes.begin(); g != tryModes.end(); ++g) {
+		if (g->bytesPerPixel != 2) {
+			g = tryModes.reverse_erase(g);
+		} else if (*g == gameDataPixelFormat()) {
```

Comment:
I don’t understand this pull request. We now get all the pixel formats which are supported by the backend, and then check if they contain the pixel format specified by the engine.

Why would we do that, and not just specify the needed pixel format in initGraphics()? The equality operator for PixelFormat checks for equality of the precision loss and left shift of each color component (a, r, g, b). So... why can’t we just specify the needed pixel format below?

Reply:
The issue with that is that if a game doesn't support the requested pixel format, the screen will remain in CLUT8 mode instead of trying to pick an alternative RGB mode. With this, if the required pixel format is available, the game will use that to reduce the amount of conversion required, but if not, it will try all available pixel formats with 2 bytes per pixel and use the first one that works.

Reply:
Thanks for the explanation:)